### PR TITLE
Implement aufile_set_position

### PR DESCRIPTION
**Pull request by larsimmisch**
This PR implements `aufile_set_position`, which sets the position in milliseconds.

### CODE REVIEW

File: rem/aufile/aufile.c [hunk 1]
```diff
@@ -278,3 +278,18 @@ size_t aufile_get_length(struct aufile *af, struct aufile_prm *prm)
...
+
+/**
+ * Set initial playing position of a WAV file in ms
+ *
+ * @param af  Audio-file
+ * @param prm Audio file parameters from aufile_open
+ *
+ * @return position in bytes or (size_t)-1 in case of an error.
+ */
+size_t aufile_set_position(struct aufile *af, struct aufile_prm *prm, size_t pos_ms)
```
Comment (cspiel1) [inline rem/aufile/aufile.c:290]: Normally in `re` all functions except getter return an error code if there are errors possible.

API functions should contain NULL pointer checks.

```c
if (!af || !prm)
    return EINVAL;
```

File: rem/aufile/aufile.c [hunk 2]
```diff
@@ -278,3 +278,18 @@ size_t aufile_get_length(struct aufile *af, struct aufile_prm *prm)
...
+{
+	size_t pos = prm->srate * aufmt_sample_size(prm->fmt) * prm->channels * pos_ms / 1000;
```
Comment (cspiel1) [inline rem/aufile/aufile.c:292]: This is only an approximation, right? Is it a good approximation for all WAV files?

Shouldn't the WAV header size be added?

Comment (larsimmisch) [reply rem/aufile/aufile.c:292]: Thanks & yes. The wav header needs to be taken into account.

File: rem/aufile/aufile.c [hunk 3]
```diff
@@ -287,9 +287,34 @@ size_t aufile_get_length(struct aufile *af, struct aufile_prm *prm)
...
-	size_t pos = prm->srate * aufmt_sample_size(prm->fmt) * prm->channels * pos_ms / 1000;
+	struct wav_fmt fmt;
+	int err;
+	size_t pos;
+
+	if (!af || !prm) {
+		return EINVAL;
+	}
+
+	// Seek to the beginning of the file
```
Comment (cspiel1) [inline rem/aufile/aufile.c]: Can be removed. It's obvious.

File: rem/aufile/aufile.c [hunk 4]
```diff
@@ -287,9 +287,34 @@ size_t aufile_get_length(struct aufile *af, struct aufile_prm *prm)
...
+	if (fseek(af->f, 0, SEEK_SET) < 0) {
+		return errno;
+	}
+	// this is only used for the side effect of moving the file ptr to the
+	// first data block.
```
Comment (cspiel1) [inline rem/aufile/aufile.c]: - Use C comments!
- Please shorter: `/* skip wav header */

File: rem/aufile/aufile.c [hunk 5]
```diff
@@ -278,3 +278,44 @@ size_t aufile_get_length(struct aufile *af, struct aufile_prm *prm)
...
+
+/**
+ * Set initial playing position of a WAV file in ms
+ *
+ * @param af  Audio-file
+ * @param prm Audio file parameters from aufile_open
+ *
+ * @return position in bytes or (size_t)-1 in case of an error.
+ */
+size_t aufile_set_position(struct aufile *af, struct aufile_prm *prm,
```
Comment (cspiel1) [inline rem/aufile/aufile.c:290]: - return int as error code
- correct  @return doxygen

Comment (larsimmisch) [reply rem/aufile/aufile.c:290]: Done

Comment (cspiel1) [reply rem/aufile/aufile.c:290]: `int` instead of `size_t`

File: rem/aufile/aufile.c [hunk 6]
```diff
@@ -278,3 +278,43 @@ size_t aufile_get_length(struct aufile *af, struct aufile_prm *prm)
...
+
+/**
+ * Set initial playing position of a WAV file in ms
+ *
+ * @param af  Audio-file
+ * @param prm Audio file parameters from aufile_open
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int aufile_set_position(struct aufile *af, struct aufile_prm *prm,
```
Comment (alfredh) [inline rem/aufile/aufile.c:290]: is `prm` modified by the function ? can it be `const` ?

File: rem/aufile/aufile.c [hunk 7]
```diff
@@ -278,3 +278,43 @@ size_t aufile_get_length(struct aufile *af, struct aufile_prm *prm)
...
+						   off_t pos_ms)
+{
+	struct wav_fmt fmt;
+	int err;
+	off_t pos;
+	size_t datasize;
+
+	if (!af || !prm) {
+		return EINVAL;
+	}
```
Comment (alfredh) [inline rem/aufile/aufile.c:300]: please remove curly braces that is not needed here ... same for the rest of the function

File: rem/aufile/aufile.c [hunk 8]
```diff
@@ -278,3 +278,43 @@ size_t aufile_get_length(struct aufile *af, struct aufile_prm *prm)
...
+ *
+ * @return 0 if success, otherwise errorcode
+ */
+int aufile_set_position(struct aufile *af, struct aufile_prm *prm,
+						   off_t pos_ms)
+{
+	struct wav_fmt fmt;
+	int err;
+	off_t pos;
+	size_t datasize;
```
Comment (alfredh) [inline rem/aufile/aufile.c:296]: please use C99 style declare+assign if possible ...

File: include/rem_aufile.h [hunk 9]
```diff
@@ -26,3 +26,5 @@ int aufile_read(struct aufile *af, uint8_t *p, size_t *sz);
 int aufile_write(struct aufile *af, const uint8_t *p, size_t sz);
 size_t aufile_get_size(struct aufile *af);
 size_t aufile_get_length(struct aufile *af, struct aufile_prm *prm);
+int aufile_set_position(struct aufile *af, struct aufile_prm *prm,
+		off_t pos_ms);
```
Comment (alfredh) [inline include/rem_aufile.h:30]: it is probably better to use `size_t` instead of `off_t` here ...

Comment (larsimmisch) [reply include/rem_aufile.h:30]: I did that first, but it's `fseek(FILE *, long, int);` and using `size_t` breaks the Windows build.

**Comment by alfredh** [comment]
my idea was to use `size_t` in the api and `off_t` in the implementation.

**Comment by larsimmisch** [comment]
Done. The cast to `off_t` was in the implementation anyway.

**Comment by alfredh** [comment]
LGTM

**Pull request merged**